DSESwift/SensorDataProxy.swift

```diff
@@ -0,0 +1,388 @@
+//
+//  SensorDataProxy.swift
+//  SensePlatform
+//
+//  Created by Tatsuya Kaneko on 15/10/15.
+//
+//
+
+import Foundation
+import Just
+
+public class SensorDataProxy {
+    
+    public enum Server {
+        case LIVE
+        case STAGING
+    }
+    
+    public enum ProxyError: ErrorType, Equatable{
+        case SessionIdNotSet
+        case InvalidSessionId
+        case InvalidSensorOrSource //thrown when invalid sensorName/sourceName is given
+        case InvalidDataFormat
+        case InvalidQuery
+        case SensorDoesNotExistOrBadStructure
+        case UnknownError
+    }
+    
+    let BASE_URL_LIVE = "https://sensor-api.sense-os.nl";
+    let BASE_URL_STAGING = "http://sensor-api.staging.sense-os.nl";
+    
+    var baseUrl: String?
+    var appKey: String?
+    var sessionId: String?
+    
+    
+    /**
+    * Create a sensor data proxy.
+    * @param server     Select whether to use the live or staging server.
+    * appKey     Application key, identifying the application in the REST API.
+    * @param sessionId  The session id of the current user.
+    */
+    init (server: Server, appKey: String, sessionId: String) {
+        self.baseUrl = (server == Server.LIVE) ? BASE_URL_LIVE : BASE_URL_STAGING;
+        self.appKey = appKey;
+        self.sessionId = sessionId;
+    }
+    
+    func setSessionId(sessionId: String){
+        self.sessionId = sessionId
+    }
+    
+    /**
+    * Get all sensor profiles
+    * Throws an exception when no sessionId is set or when the sessionId is not valid.
+    * @return Returns an array with sensor profiles, structured like:
+    *         [{sensor_name: string, data_type: JSON}, ...]
+    */
+    func getSensorProfiles() throws -> Array<AnyObject>?{
+        if (!isSessionIdSet()){ throw ProxyError.SessionIdNotSet }
+        
+        let result = Just.get(self.baseUrl! + "/sensor_profiles", headers: getHeaders())
+        if let error = checkStatusCode(result.statusCode!, successfulCode: 200){
```

Comment:
@kanekin there's possibility that result.statusCode is nil if there is a networking problem. i think this should be check before extracting the value by testing if statusCode != nil, or testing if result.ok